plugins/pulp_ostree/plugins/importers/steps.py

```diff
@@ -147,40 +139,32 @@ def __init__(self):
         super(Add, self).__init__(step_type=constants.IMPORT_STEP_ADD_UNITS)
         self.description = _('Add Content Units')
 
-    def process_main(self):
+    def process_main(self, item=None):
         """
         Find all branch (heads) in the local repository and
-        create content units for them.
+        create content units for them.  The
         """
-        conduit = self.get_conduit()
-        repository = lib.Repository(self.parent.storage_path)
-        for ref in repository.list_refs():
+        repository = Repository.objects.get(repo_id=self.get_repo().id)
+        lib_repository = lib.Repository(self.parent.storage_dir)
+        for ref in lib_repository.list_refs():
             if ref.path not in self.parent.branches:
                 # not listed
                 continue
-            commit = model.Commit(ref.commit, ref.metadata)
-            unit = model.Unit(self.parent.remote_id, ref.path, commit)
-            self.link(unit)
-            _unit = Unit(constants.OSTREE_TYPE_ID, unit.key, unit.metadata, unit.storage_path)
-            conduit.save_unit(_unit)
-
-    def link(self, unit):
-        """
-        Link the unit storage path to the main *content* storage path.
-        The link will be verified if it already exits.
-
-        :param unit: The unit to linked.
-        :type unit: model.Unit
-        """
-        link = unit.storage_path
-        target = self.parent.storage_path
-        try:
-            os.symlink(target, link)
-        except OSError, e:
-            if e.errno == errno.EEXIST and os.path.islink(link) and os.readlink(link) == target:
-                pass  # identical
-            else:
-                raise
+            unit = model.Branch(
+                remote_id=self.parent.remote_id,
+                branch=ref.path,
+                commit=ref.commit,
+                metadata=ref.metadata)
+            try:
+                unit.save()
+            except NotUniqueError:
+                # ignored
+                pass
+            try:
+                associate_single_unit(repository, unit)
```

Comment:
1) If the previous save failed because the unit was not unique, this will fail because the unit you have here does not have the _id value used to correlate the repo & the unit.
2) This will never raise a NotUniqueError as the existing association is updated if the unit has already been associated to the repo.